ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/HiveAggregateJoinTransposeRule.java

```diff
@@ -290,7 +291,8 @@ public void onMatch(RelOptRuleCall call) {
       RelNode r = relBuilder.build();
       RelOptCost afterCost = mq.getCumulativeCost(r);
       RelOptCost beforeCost = mq.getCumulativeCost(aggregate);
-      if (afterCost.isLt(beforeCost)) {
```

Comment:
I think you suggested changing this... Maybe `isLe` if we do not introduce an additional aggregate on top?

Reply:
yes; if we use `isLe` the current cost model which only takes rowcount into account will prefer the pushing aggregates further

there is another alternative to the `force` based approach: the rule can be configured to use the more advanced cost system - so that it could take cpu/io cost into account